dali/pipeline/graph/op_graph2.cc

```diff
@@ -176,12 +176,17 @@ class OpGraph::SortHelper {
     // NOTE: no producer of otherwise valid node should be removed, so we only need to adjust
     // consumers.
 
-    for (auto &pruned_op : out_ops) {
-      graph_.RemoveDataNodeReferences(pruned_op);
-      graph_.name2op_.erase(pruned_op.instance_name);
-    }
-    for (auto &pruned_data : out_data) {
-      graph_.name2data_.erase(pruned_data.name);
+    if (prune) {
+      for (auto &pruned_op : out_ops) {
+        graph_.RemoveDataNodeReferences(pruned_op);
+        graph_.name2op_.erase(pruned_op.instance_name);
+      }
+      for (auto &pruned_data : out_data) {
+        graph_.name2data_.erase(pruned_data.name);
+      }
+    } else {
+      graph_.op_nodes_.splice(graph_.op_nodes_.end(), out_ops);
+      graph_.data_nodes_.splice(graph_.data_nodes_.end(), out_data);
```

Comment:
Just a note, you didn't traverse the nodes that are not reachable from the outputs so they may be listed here in non-topological order.

Reply:
Nice catch. Perhaps I'll just traverse everything when not pruning and then skip this splicing.

Reply:
Fixed.